include/swift/SILOptimizer/Analysis/ARCAnalysis.h

```diff
@@ -221,6 +221,18 @@ class ConsumedArgToEpilogueReleaseMatcher {
 
   bool hasBlock() const { return HasBlock; }
 
+  bool isEpilogueRelease(SILInstruction *I) const {
```

Comment:
Prior to looping through the instructions you should be able to get the exit block and see if I->getBlock() is the same block.For example you can refactor the code in ::recompute() to separate out the switch on ExitKind and selection of return/throw block, and then test against that.